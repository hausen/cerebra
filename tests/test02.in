OBS:X,Z,W,Y
EXO:U1,U2,U3,U4,A,B,C

U1 = .5
U2 = .5
U3 = .5
U4 = .5
A  = .5
B  = .5
C  = .5

X = U1
Z = U1 + U2 
W = Z 
W = ((U3) | B) + (B & (1 + Z))    
Y = X + W + U2
Y = ((X + W + U2) & A) + (A | (U4 + X + W + U2))

OUTPUT 1 : "versao uncondicional"
        MARGIN: X, Y, W, Z
        INTERV:

OUTPUT 2 : "interv. tipo do(X=1)"
        MARGIN: X, Y
        INTERV: X = 1

OUTPUT 3 : "interv. tipo do(Z=1)"
        MARGIN: X, W, Y
        INTERV: Z = 1 

OUTPUT 4 : "interv. tipo do(Z=0)"
        MARGIN: X, W, Y
        INTERV: Z = 0

OUTPUT 5 : "P(X)"
        MARGIN: X
        INTERV:

OUTPUT 6 : "P(Z)"
        MARGIN: Z
        INTERV:

OUTPUT 7 : "P(Z, W)"
        MARGIN: Z, W
        INTERV:

OUTPUT 8 : "P(X, Z, Y)"
        MARGIN: X, Z, Y
        INTERV:
