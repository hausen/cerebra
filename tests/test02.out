OUTPUT 1: "versao uncondicional"
X	Y	W	Z	Prob.
0	0	0	0	0.0468750
0	0	0	1	0.0468750
0	0	1	0	0.0468750
0	0	1	1	0.0468750
0	1	0	0	0.1406250
0	1	0	1	0.0156250
0	1	1	0	0.0156250
0	1	1	1	0.1406250
1	0	0	0	0.0468750
1	0	0	1	0.0468750
1	0	1	0	0.0468750
1	0	1	1	0.0468750
1	1	0	0	0.1406250
1	1	0	1	0.0156250
1	1	1	0	0.0156250
1	1	1	1	0.1406250
END OUTPUT

OUTPUT 2: "interv. tipo do(X=1)"
X	Y	Prob.
0	0	0.1875000
0	1	0.3125000
1	0	0.1875000
1	1	0.3125000
END OUTPUT

OUTPUT 3: "interv. tipo do(Z=1)"
X	W	Y	Prob.
0	0	0	0.0937500
0	0	1	0.1562500
0	1	0	0.0937500
0	1	1	0.1562500
1	0	0	0.0937500
1	0	1	0.1562500
1	1	0	0.0937500
1	1	1	0.1562500
END OUTPUT

OUTPUT 4: "interv. tipo do(Z=0)"
X	W	Y	Prob.
0	0	0	0.0937500
0	0	1	0.1562500
0	1	0	0.0937500
0	1	1	0.1562500
1	0	0	0.0937500
1	0	1	0.1562500
1	1	0	0.0937500
1	1	1	0.1562500
END OUTPUT

OUTPUT 5: "P(X)"
X	Prob.
0	0.5000000
1	0.5000000
END OUTPUT

OUTPUT 6: "P(Z)"
Z	Prob.
0	0.5000000
1	0.5000000
END OUTPUT

OUTPUT 7: "P(Z, W)"
Z	W	Prob.
0	0	0.3750000
0	1	0.1250000
1	0	0.1250000
1	1	0.3750000
END OUTPUT

OUTPUT 8: "P(X, Z, Y)"
X	Z	Y	Prob.
0	0	0	0.0937500
0	0	1	0.1562500
0	1	0	0.0937500
0	1	1	0.1562500
1	0	0	0.0937500
1	0	1	0.1562500
1	1	0	0.0937500
1	1	1	0.1562500
END OUTPUT

